From 6fb888d53bdc0cffac2bf24d796367bd3927991c Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 27 May 2019 15:27:57 +0800
Subject: [PATCH] systemd: fix avc denials for systemd_sessions_t

Fix:
type=AVC msg=audit(1558940783.641:53): avc:  denied  { search } for
pid=223 comm="systemd-user-se" name="kernel" dev="proc" ino=9668
scontext=system_u:system_r:systemd_sessions_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=dir permissive=1
type=AVC msg=audit(1558940783.641:53): avc:  denied  { read } for
pid=223 comm="systemd-user-se" name="osrelease" dev="proc" ino=10403
scontext=system_u:system_r:systemd_sessions_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=file permissive=1
type=AVC msg=audit(1558940783.641:53): avc:  denied  { open } for
pid=223 comm="systemd-user-se" path="/proc/sys/kernel/osrelease"
dev="proc" ino=10403 scontext=system_u:system_r:systemd_sessions_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=file permissive=1
type=AVC msg=audit(1558940783.641:54): avc:  denied  { getattr } for
pid=223 comm="systemd-user-se" path="/proc/sys/kernel/osrelease"
dev="proc" ino=10403 scontext=system_u:system_r:systemd_sessions_t
tcontext=system_u:object_r:sysctl_kernel_t tclass=file permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index fb2719b..3cf4002 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1079,6 +1079,8 @@ seutil_read_file_contexts(systemd_sessions_t)
 systemd_log_parse_environment(systemd_sessions_t)
 seutil_dontaudit_read_file_contexts(systemd_sessions_t)
 
+kernel_read_kernel_sysctls(systemd_sessions_t)
+
 #########################################
 #
 # Tmpfiles local policy
-- 
2.7.4

